internal/compiler/program.go

```diff
@@ -985,41 +1031,29 @@ func (p *Program) GetGlobalDiagnostics(ctx context.Context) []*ast.Diagnostic {
 		return nil
 	}
 
-	globalDiagnostics := make([][]*ast.Diagnostic, p.checkerPool.Count())
-	p.checkerPool.ForEachCheckerParallel(ctx, func(idx int, checker *checker.Checker) {
+	pool := p.checkerPool.(*checkerPool)
```

Comment:
Why the interface and struct implementation if we're always going to cast to a concrete instance in these methods?